CodeEdit/Features/DebugArea/DebugAreaView.swift

```diff
@@ -7,33 +7,64 @@
 
 import SwiftUI
 
+struct AreaTab: Identifiable, Equatable {
+    var id: String
+    var title: String
+    var systemImage: String? = "e.square"
+    var contentView: () -> AnyView
+
+    init<Content: View>(id: String, title: String, systemImage: String? = "e.square", @ViewBuilder content: @escaping () -> Content) {
+        self.id = id
+        self.title = title
+        self.systemImage = systemImage
+        self.contentView = { AnyView(content()) }
+    }
```

Comment:
The use of `AnyView` is discouraged, and should only be used as a last resort.

Reply:
Using
```swift
struct AreaTab<Content: View>: Identifiable, Equatable {
  // ...
  var contentView: () -> Content
  // ...
  init(/*params here*/) {
    // ...
    self.contentView = content
  }
  // ...
}
```

is better practice

Reply:
> Using
> 
> ```swift
> struct AreaTab<Content: View>: Identifiable, Equatable {
>   // ...
>   var contentView: () -> Content
>   // ...
>   init(/*params here*/) {
>     // ...
>     self.contentView = content
>   }
>   // ...
> }
> ```
> 
> is better practice

This probably won't work though, as it would require storing structs with different associated types (different `Content` types)

Reply:
Oh, I just realised that this struct contains both a content view and the tab's data. What's the rationale behind that choice? Usually, the data is passed to a view for the view to display the data, they aren't bundled together. Storing views isn't the best idea.

Reply:
> Oh, I just realised that this struct contains both a content view and the tab's data. What's the rationale behind that choice? Usually, the data is passed to a view for the view to display the data, they aren't bundled together. Storing views isn't the best idea.

I get the rationale behind putting the view with the data though, as they're kind of tied to each other. I've put back the enums that were there before the PR, and I've made a custom protocol which is used by `AreaTabBar`. This allows for a similar kind of "grouping" the data, while not needing to use AnyView:
```swift
enum AreaTabTest: AreaBarIcon, CaseIterable {
    var id: Self { self }

    case terminal
    case debugConsole
    case output

    var title: String {
        switch self {
        case .terminal:
            return "Terminal"
        case .debugConsole:
            return "Debug Console"
        case .output:
            return "Output"
        }
    }

    var systemImage: String {
        switch self {
        case .terminal:
            return "terminal"
        case .debugConsole:
            return "ladybug"
        case .output:
            return "list.bullet.indent"
        }
    }

    var body: some View {
        switch self {
        case .terminal:
            DebugAreaTerminalView()
        case .debugConsole:
            DebugAreaDebugView()
        case .output:
            DebugAreaOutputView()
        }
    }
}
```